test/itkSpectra1DAveragingImageFilterTest.cxx

```diff
@@ -0,0 +1,188 @@
+  itk::IdentifierType testPixelCount = test->GetLargestPossibleRegion().GetNumberOfPixels();
+  if (pixelCount != testPixelCount)
+  {
+    itkGenericExceptionMacro(<< "Images have different number of pixels, " << pixelCount << " vs " << testPixelCount);
+  }
+
+  itk::IdentifierType differentPixels = 0;
+  itk::IdentifierType linePixels = baseline->GetLargestPossibleRegion().GetSize()[0];
+  itk::IdentifierType lineCount = pixelCount / linePixels;
+
+  using InputPixelType1 = typename TInImage1::PixelType;
+  using InputPixelType2 = typename TInImage2::PixelType;
+  auto * bufferBaseline = static_cast<InputPixelType1 *>(baseline->GetBufferPointer());
+  auto * bufferTest = static_cast<InputPixelType2 *>(test->GetBufferPointer());
+  for (itk::IdentifierType p = 0; p < pixelCount; ++p)
+  {
+    for (unsigned d = 0; d < TInImage1::PixelType::Dimension; ++d)
+    {
+      if (!itk::Math::FloatAlmostEqual(bufferBaseline[p][d], bufferTest[p][d], lineCount, 1e-4))
+      {
+        ++differentPixels;
+        std::cout << "Images differ at " << p << "[" << d << "]: " << bufferBaseline[p][d] << " vs " << bufferTest[p][d]
+                  << "\n";
+      }
+    }
+  }
+
+  itk::IdentifierType maxDifferentPixels = 0;
+  if (differentPixels > maxDifferentPixels)
+  {
+    itkGenericExceptionMacro(<< differentPixels << " pixels are different, maximum allowed is " << maxDifferentPixels);
+  }
+}
+
+template <typename TInImage, typename TOutImage>
+int
+doTest(int argc, char * argv[], itk::SmartPointer<TOutImage> & output)
+{
+  using AverageFilterType = itk::Spectra1DAveragingImageFilter<TInImage, TOutImage>;
+  typename AverageFilterType::Pointer averageFilter = AverageFilterType::New();
+
+  itk::SimpleFilterWatcher watcher(averageFilter);
+
+  for (int i = 2; i < argc; ++i)
+  {
+    using ReaderType = itk::ImageFileReader<TInImage>;
+    typename ReaderType::Pointer reader = ReaderType::New();
+    reader->SetFileName(argv[i]);
+    averageFilter->SetInput(i - 2, reader->GetOutput());
+
+    // it should not be necessary to do this here, the filter is supposed to call this when it needs this input
+    ITK_TRY_EXPECT_NO_EXCEPTION(reader->Update());
+  }
+
+  ITK_TRY_EXPECT_NO_EXCEPTION(averageFilter->Update());
+  output = averageFilter->GetOutput();
+  return EXIT_SUCCESS;
+}
+
+template <typename TInImage, typename TOutImage>
+itk::SmartPointer<TOutImage>
+createReference(int argc, char * argv[])
+{
+  using AverageFilterType = itk::Spectra1DAveragingImageFilter<TInImage, TOutImage>;
+  auto averageFilter = AverageFilterType::New();
+
+  for (int i = 2; i < argc; ++i)
+  {
+    using ReaderType = itk::ImageFileReader<TInImage>;
+    typename ReaderType::Pointer reader = ReaderType::New();
+    reader->SetFileName(argv[i]);
+    averageFilter->SetInput(i - 2, reader->GetOutput());
+
+    // it should not be necessary to do this here, the filter is supposed to call this when it needs this input
+    reader->Update();
```

Comment:
Per this comment, can the `reader->Update()` statement be removed? Or is the `Update` actually required?

Reply:
Unfortunately, this call cannot be removed. The comment above expresses my feeling, but I haven't gotten to the bottom of it. I don't know what we should change in the filter to not require this `reader->Update()` call here.

Reply:
Seems fine to leave it in for now. Thanks.

Reply:
Maybe @thewtex has some suggestions?